src/main/java/HttpHandlerSelector.java

```diff
@@ -1,114 +1,24 @@
     if (request.method().equals("GET")) {
-      return getRequest(request);
-    }
-    return notFound(request);
-  }
-
-  public HttpResponse generate400Response() {
-    byte[] body = "400 : Bad Request".getBytes();
-    return new HttpResponse(CODE_400, body, buildHeaders(body, HTML));
-  }
-
-  private HttpResponse getRequest(HttpRequest request) {
-    String path = request.path();
-    if (path.equals("/")) {
-      return defaultResponse();
-    }
-    try {
-      byte[] body = fileIO.getContent(path);
-      HashMap<String, String> headers = buildHeaders(body, path);
-      return new HttpResponse(CODE_200, body, headers);
-    }
-    catch (IOException e) {
-      return notFound(request);
+      return getRequest.generateResponse(request);
     }
-  }
-
-  private HttpResponse defaultResponse() {
-    byte[] body = "Default path / ".getBytes();
-    HashMap<String, String> headers = buildHeaders(body, "/");
-    return new HttpResponse(CODE_200, body, headers);
-  }
 
-  private HttpResponse notFound(HttpRequest request) {
-    byte[] body = "File not Found".getBytes();
-    HashMap<String, String> headers = buildHeaders(body, request.path());
-    return new HttpResponse(CODE_404, body, headers);
-  }
-
-  private HashMap<String, String> buildHeaders(byte[] body, String path) {
-    HashMap<String, String> rtnMap = new HashMap<String, String>();
-    rtnMap.putAll(standardHeaders);
-    rtnMap.put("Date", getDateTime());
-    rtnMap.putAll(contentHeaders(body, path));
-    return rtnMap;
-  }
-
-  private String getDateTime() {
-    Date date = new Date();
-    DateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd hh:mm:ss");
-    String strDate = dateFormat.format(date);
-    return strDate;
-  }
-
-  private HashMap<String, String> contentHeaders(byte[] body, String path) {
-    HashMap<String, String> rtnMap = new HashMap<String, String>();
-    rtnMap.put("Content-Type", contentType(path));
-    rtnMap.put("Content-Length", contentLength(body));
-    return rtnMap;
-  }
-
-  private String contentDisposition(String fileName) {
-    if (fileName.endsWith(".jpg") || fileName.endsWith(".jpeg")) {
-      return "inline";
-    }
-    return " ";
-  }
-  private String contentLength(byte[] body) {
-    return Integer.toString(body.length);
-  }
-
-  private String contentType(String fileName) {
-    if (fileName.endsWith(".htm") || fileName.endsWith(".html")
-        || fileName.endsWith(".txt")) {
-      return "text/html";
-    } else if (fileName.endsWith(".jpg") || fileName.endsWith(".jpeg")) {
-      return "image/jpeg";
-    } else if (fileName.endsWith(".gif")) {
-      return "image/gif";
-    } else if (fileName.endsWith(".pdf")) {
-      return "application/pdf";
-    } else {
-      return "text/html";
-    }
+    // TODO :: Remove this after adding classes to handle other types of request methods
+    String responseMessage = "This server currenlty doesn't support request type " + request.method() + "\r\n";
```

Comment:
yes, it probably should. I suppose since I knew this would get changed later I didn't bother to do it properly. Which is a bad habit I need to fix. As I'm sure there will be moments in production where code that you might assume will get changed later, won't actually get changed.